Libraries/LibJS/AST.cpp

```diff
@@ -82,12 +90,14 @@ Value CallExpression::execute(Interpreter& interpreter) const
     }
 
     Object* new_object = nullptr;
+    Value result;
     if (is_new_expression()) {
         new_object = interpreter.heap().allocate<Object>();
         auto prototype = function->get("prototype");
         if (prototype.has_value() && prototype.value().is_object())
             new_object->set_prototype(prototype.value().as_object());
         call_frame.this_value = new_object;
+        result = function->construct(interpreter);
```

Comment:
That's not right, you have to replace the result with the new object according to those rules (if it's a primitive or something).

Reply:
I didn't change that, it's still happening below:

https://github.com/linusg/serenity/blob/a9a3ba51d27ebd8816b7e7893e48e727ff219eba/Libraries/LibJS/AST.cpp#L116-L121